File: festival-web/src/main/java/cn/bdqfork/web/annotation/VerticleMapping.java
package cn.bdqfork.web.annotation;

import javax.inject.Named;
import javax.inject.Singleton;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * 该注解用于将服务转化为Verticle，服务之间的调用将通过EventBus进行通信。
 *
 * @author bdq
 * @since 2020/1/21
 */
@Singleton
@Named
@Documented
@Retention(RUNTIME)
@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE})
public @interface VerticleMapping {
    String value() default "";
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/annotation/VerticleMapping.kt
package cn.bdqfork.kotlin.web.annotation

import javax.inject.Named
import javax.inject.Singleton

/**
 * 该注解用于将服务转化为Verticle，服务之间的调用将通过EventBus进行通信。
 *
 * @author bdq
 * @since 2020/1/21
 */
@Singleton
@Named
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class VerticleMapping(val value: String = "")